### ISSUE TITLE
Agent's test_cluster.py:test_auto_sync "404 Unexpected Error: Not Found for uri"

### ISSUE THREAD

**Issue by ssorenso** [F5Networks/f5-openstack-agent#877]
* Title: Agent's test_cluster.py:test_auto_sync "404 Unexpected Error: Not Found for uri"
* Attachments:
* Details: 

##### Suggested Error Type:
Bug... this may be an SDK issue

##### Traceback:
Liberty:
```
mgmt_root = <f5.bigip.ManagementRoot object at 0x7f46a4a6f650>

    def test_auto_sync(mgmt_root):
        # assume BIG-IP has device group test-group
        device_group = 'test-group'
>       cm.disable_auto_sync(device_group, mgmt_root)

test_cluster.py:54:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5_openstack_agent/lbaasv2/drivers/bigip/cluster_manager.py:38: in disable_auto_sync
    name=device_group_name, partition=partition)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:1026: in load
    return self._load(**kwargs)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:1005: in _load
    response = refresh_session.get(base_uri, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <icontrol.session.iControlRESTSession object at 0x7f46a4a6ff50>
RIC_base_uri = 'https://10.190.24.88:443/mgmt/tm/cm/device-group/', kwargs = {}
partition = 'Common', sub_path = '', suffix = '', identifier = 'test-group'
uri_as_parts = True, transform_name = False
REST_uri = 'https://10.190.24.88:443/mgmt/tm/cm/device-group/~Common~test-group'

    @functools.wraps(method)
    def wrapper(self, RIC_base_uri, **kwargs):
        partition = kwargs.pop('partition', '')
        sub_path = kwargs.pop('subPath', '')
        suffix = kwargs.pop('suffix', '')
        identifier, kwargs = _unique_resource_identifier_from_kwargs(**kwargs)
        uri_as_parts = kwargs.pop('uri_as_parts', False)
        transform_name = kwargs.pop('transform_name', False)
        if uri_as_parts:
            REST_uri = generate_bigip_uri(RIC_base_uri, partition, identifier,
                                          sub_path, suffix,
                                          transform_name=transform_name,
                                          **kwargs)
        else:
            REST_uri = RIC_base_uri
        pre_message = "%s WITH uri: %s AND suffix: %s AND kwargs: %s" %\
            (method.__name__, REST_uri, suffix, kwargs)
        logging.debug(pre_message)
        response = method(self, REST_uri, **kwargs)
        post_message =\
            "RESPONSE::STATUS: %s Content-Type: %s Content-Encoding:"\
            " %s\nText: %r" % (response.status_code,
                               response.headers.get('Content-Type', None),
                               response.headers.get('Content-Encoding', None),
                               response.text)
        logging.debug(post_message)
        if response.status_code not in range(200, 207):
            error_message = '%s Unexpected Error: %s for uri: %s\nText: %r' %\
                            (response.status_code,
                             response.reason,
                             response.url,
                             response.text)
>           raise iControlUnexpectedHTTPError(error_message, response=response)
E           iControlUnexpectedHTTPError: 404 Unexpected Error: Not Found for uri: https://10.190.24.88:443/mgmt/tm/cm/device-group/~Common~test-group
E           Text: u'{"code":404,"message":"01020036:3: The requested device group (/Common/test-group) was not found.","errorStack":[],"apiError":3}'

/usr/local/lib/python2.7/dist-packages/icontrol/session.py:272: iControlUnexpectedHTTPError
```
Mitaka:
```
mgmt_root = <f5.bigip.ManagementRoot object at 0x7f3591724d90>

    def test_auto_sync(mgmt_root):
        # assume BIG-IP has device group test-group
        device_group = 'test-group'
>       cm.disable_auto_sync(device_group, mgmt_root)

test_cluster.py:54:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5_openstack_agent/lbaasv2/drivers/bigip/cluster_manager.py:38: in disable_auto_sync
    name=device_group_name, partition=partition)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:1026: in load
    return self._load(**kwargs)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:1005: in _load
    response = refresh_session.get(base_uri, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <icontrol.session.iControlRESTSession object at 0x7f35917244d0>
RIC_base_uri = 'https://10.190.24.89:443/mgmt/tm/cm/device-group/', kwargs = {}
partition = 'Common', sub_path = '', suffix = '', identifier = 'test-group'
uri_as_parts = True, transform_name = False
REST_uri = 'https://10.190.24.89:443/mgmt/tm/cm/device-group/~Common~test-group'

    @functools.wraps(method)
    def wrapper(self, RIC_base_uri, **kwargs):
        partition = kwargs.pop('partition', '')
        sub_path = kwargs.pop('subPath', '')
        suffix = kwargs.pop('suffix', '')
        identifier, kwargs = _unique_resource_identifier_from_kwargs(**kwargs)
        uri_as_parts = kwargs.pop('uri_as_parts', False)
        transform_name = kwargs.pop('transform_name', False)
        if uri_as_parts:
            REST_uri = generate_bigip_uri(RIC_base_uri, partition, identifier,
                                          sub_path, suffix,
                                          transform_name=transform_name,
                                          **kwargs)
        else:
            REST_uri = RIC_base_uri
        pre_message = "%s WITH uri: %s AND suffix: %s AND kwargs: %s" %\
            (method.__name__, REST_uri, suffix, kwargs)
        logging.debug(pre_message)
        response = method(self, REST_uri, **kwargs)
        post_message =\
            "RESPONSE::STATUS: %s Content-Type: %s Content-Encoding:"\
            " %s\nText: %r" % (response.status_code,
                               response.headers.get('Content-Type', None),
                               response.headers.get('Content-Encoding', None),
                               response.text)
        logging.debug(post_message)
        if response.status_code not in range(200, 207):
            error_message = '%s Unexpected Error: %s for uri: %s\nText: %r' %\
                            (response.status_code,
                             response.reason,
                             response.url,
                             response.text)
>           raise iControlUnexpectedHTTPError(error_message, response=response)
E           iControlUnexpectedHTTPError: 404 Unexpected Error: Not Found for uri: https://10.190.24.89:443/mgmt/tm/cm/device-group/~Common~test-group
E           Text: u'{"code":404,"message":"01020036:3: The requested device group (/Common/test-group) was not found.","errorStack":[],"apiError":3}'

/usr/local/lib/python2.7/dist-packages/icontrol/session.py:272: iControlUnexpectedHTTPError
```
Newton:
```
mgmt_root = <f5.bigip.ManagementRoot object at 0x7fe3d41cbe10>

    def test_auto_sync(mgmt_root):
        # assume BIG-IP has device group test-group
        device_group = 'test-group'
>       cm.disable_auto_sync(device_group, mgmt_root)

test_cluster.py:54:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5_openstack_agent/lbaasv2/drivers/bigip/cluster_manager.py:38: in disable_auto_sync
    name=device_group_name, partition=partition)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:1026: in load
    return self._load(**kwargs)
../../../.tox/singlebigip/local/lib/python2.7/site-packages/f5/bigip/resource.py:1005: in _load
    response = refresh_session.get(base_uri, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <icontrol.session.iControlRESTSession object at 0x7fe3d41cb4d0>
RIC_base_uri = 'https://10.190.25.10:443/mgmt/tm/cm/device-group/', kwargs = {}
partition = 'Common', sub_path = '', suffix = '', identifier = 'test-group'
uri_as_parts = True, transform_name = False
REST_uri = 'https://10.190.25.10:443/mgmt/tm/cm/device-group/~Common~test-group'

    @functools.wraps(method)
    def wrapper(self, RIC_base_uri, **kwargs):
        partition = kwargs.pop('partition', '')
        sub_path = kwargs.pop('subPath', '')
        suffix = kwargs.pop('suffix', '')
        identifier, kwargs = _unique_resource_identifier_from_kwargs(**kwargs)
        uri_as_parts = kwargs.pop('uri_as_parts', False)
        transform_name = kwargs.pop('transform_name', False)
        if uri_as_parts:
            REST_uri = generate_bigip_uri(RIC_base_uri, partition, identifier,
                                          sub_path, suffix,
                                          transform_name=transform_name,
                                          **kwargs)
        else:
            REST_uri = RIC_base_uri
        pre_message = "%s WITH uri: %s AND suffix: %s AND kwargs: %s" %\
            (method.__name__, REST_uri, suffix, kwargs)
        logging.debug(pre_message)
        response = method(self, REST_uri, **kwargs)
        post_message =\
            "RESPONSE::STATUS: %s Content-Type: %s Content-Encoding:"\
            " %s\nText: %r" % (response.status_code,
                               response.headers.get('Content-Type', None),
                               response.headers.get('Content-Encoding', None),
                               response.text)
        logging.debug(post_message)
        if response.status_code not in range(200, 207):
            error_message = '%s Unexpected Error: %s for uri: %s\nText: %r' %\
                            (response.status_code,
                             response.reason,
                             response.url,
                             response.text)
>           raise iControlUnexpectedHTTPError(error_message, response=response)
E           iControlUnexpectedHTTPError: 404 Unexpected Error: Not Found for uri: https://10.190.25.10:443/mgmt/tm/cm/device-group/~Common~test-group
E           Text: u'{"code":404,"message":"01020036:3: The requested device group (/Common/test-group) was not found.","errorStack":[],"apiError":3}'

/usr/local/lib/python2.7/dist-packages/icontrol/session.py:272: iControlUnexpectedHTTPError
```

##### Impacted Tests:
* `openstack_agent_liberty_12.1.2-overcloud.6.consoleText`
  * Type: failures
  * Test: `test_auto_sync`
  * Sum: `md5sum:27c102f448d62c5ccfcfc6b762fff478`
* `openstack_agent_newton_12.1.2-overcloud.6.consoleText`
  * Type: failures
  * Test: `test_auto_sync`
  * Sum: `md5sum:90a2ac24af915aaf7df750189a98af0d`
* `openstack_agent_mitaka_12.1.2-overcloud.6.consoleText`
  * Type: failures
  * Test: `test_auto_sync`
  * Sum: `md5sum:f129804f74d0376e5a2d27d8dd8c84ee`

#### OpenStack Release
All supported

#### Agent Version
* liberty - `bc50989f5c75da211189c03351aca7cd03e67ccf`
* mitaka - `b096048814d641ff27ddc0be0babf70a6a6d230d`
* newton - `2060bb6215e25a86e5dfdfebf42e1f6c5ad5d464`

#### Operating System
Nightly

#### Deployment
singlebigip newton

**Comment by pjbreaux** [comment]
@dflanigan @mattgreene @ssorenso: I don't believe these to be bugs, but test issues. All of the three tests are predicated on the fact that there exists a device group named 'test-group', but I see nothing in the test setup to create such a thing before these tests start (unless it's somewhere else). There is a default group on the bigip devices, but it is called 'device_trust_group'. Anyway, looks like a test setup issue.

**Comment by richbrowne** [comment]
Tests were fixed as part of 0cdebc5be80dc4ba8759672f6ee808123c590e5e
